### PR TITLE
Java: don't use List<ResourceName> in flattened methods

### PR DESCRIPTION
**Pull request by andreamlin**
Flattened methods with List<ResourceName> as a param will have the same erasure as the same flattened RPC method with List<String> as a param, and this is a compile error.

Don't use ResourceNames when a fieldconfig is for a repeated field.

In google-cloud-logging, this PR provides the following bug fix: https://github.com/andreamlin/google-cloud-java/commit/386c987bf831c34377e08a3e2638eb5d5ff5385b

### CODE REVIEW

File: src/test/java/com/google/api/codegen/testdata/java/java_library.baseline [hunk 1]
```diff
@@ -2363,35 +2362,6 @@ public class LibraryClient implements BackgroundResource {
...
-   *     // doThingsWith(element);
-   *   }
-   * }
-   * </code></pre>
-   *
-   * @param names
-   * @param shelves
-   * @throws com.google.api.gax.rpc.ApiException if the remote call fails
-   */
-  public final FindRelatedBooksPagedResponse findRelatedBooks(List<ShelfBookName> names, List<ShelfName> shelves) {
```
Comment (andreamlin) [inline src/test/java/com/google/api/codegen/testdata/java/java_library.baseline]: This flattened method has the same erasure as `FindRelatedBooksPagedResponse findRelatedBooks(List<String> names, List<String> shelves)` on line 2415 of the diffbase version.

**Comment by andreamlin** [comment]
PTAL

File: src/test/java/com/google/api/codegen/testdata/java/java_library.baseline [hunk 2]
```diff
@@ -3128,9 +3098,9 @@ public class LibraryClient implements BackgroundResource {
    *   String requiredSingularString = "";
    *   ByteString requiredSingularBytes = ByteString.copyFromUtf8("");
    *   TestOptionalRequiredFlatteningParamsRequest.InnerMessage requiredSingularMessage = TestOptionalRequiredFlatteningParamsRequest.InnerMessage.newBuilder().build();
-   *   ShelfBookName requiredSingularResourceName = ShelfBookName.of("[SHELF_ID]", "[BOOK_ID]");
-   *   BookName requiredSingularResourceNameOneof = ShelfBookName.of("[SHELF_ID]", "[BOOK_ID]");
-   *   ProjectName requiredSingularResourceNameCommon = ProjectName.of("[PROJECT]");
+   *   String requiredSingularResourceName = "";
```
Comment (garrettjonesgoogle) [inline src/test/java/com/google/api/codegen/testdata/java/java_library.baseline]: The change to the singular field here doesn't seem right.

**Comment by andreamlin** [comment]
Addressed comments; PTAL

**Comment by andreamlin** [comment]
Added issue #2137 in the comments of the diff.

**Comment by andreamlin** [comment]
PTAL

**Pull request merged**